### ISSUE TITLE
Add a diagram showing how things in the model relate to each other

### ISSUE THREAD

**Issue by marcoscaceres** [w3c-fedid/digital-credentials#66]
It would be nice to have a diagram showing how parts of the model relate to each other.

**Comment by OR13** [comment]
I can make a mermaid.

How much detail are you looking for?

**Comment by marcoscaceres** [comment]
Not much... just enough to give a simple overview of how things related to each other and maybe super simple example of a credential picker (super abstract though... like just a box showing a driver's-license-like credential). 

We might be able to make it more complex from there, showing how things are verified etc. and what role each thing in the model plays.

**Comment by OR13** [comment]
Here is a quick sketch... Obviously the method names would need to align to the webIDL.

```mermaid
flowchart LR

    subgraph mobile_device
        
        credential_private_key
        credential_private_key --> public_key_credential

        subgraph public_key_credential[Mobile Driver's License]
            credential_public_key
        end
    end

    subgraph user_agent
        credential_manager[Select Digital Credential]
    end

    subgraph relying_party
        request_proof_of_age[Request Age Verification]
    end

relying_party -.- presentationRequest -.-> user_agent
relying_party -.- presentationResponse -.- user_agent

user_agent -.- mdocRequest -.-> mobile_device

user_agent -.- mdocResponse -.- mobile_device


```

**Comment by marcoscaceres** [comment]
Ok, awesome! This is a great start. 

So yeah, once we land #57 we can start adding some of the terminology from there to the diagram... we also don't yet have any mention of mDoc (#67) yet, so we can incrementally build this up as we add things to the spec. 

This is already hugely helpful though.

**Comment by timcappalli** [comment]
> Green arrows seem to be controlled by Chromium, Safari, Geko (Browsers)
>
> Purple arrows seem to be controlled by iOS and Android / Samsung. (Mobile OS)
> 
> Red arrows seem to be controlled by macOS / linux / windows (Desktop OS)
> 
> Blue arrows seem to be controlled by verifier protocols (OpenID specs, mDoc specs).

Green arrows in the browser would be implemented by browser engines: Chromium, Webkit, Geck, etc

Green arrows between devices would be implemented by the client platforms: Android, Windows, macOS, iOS, Linux (with some small exceptions where browsers may handle it themselves)

Purple and Red arrows are implemented by app platforms: Android, Windows, macOS, iOS, Linux

**Comment by marcoscaceres** [comment]
This depends on #83 (that is to say, whatever ends up in #83 should be the words used in the diagram... along with "platform" etc.)

**Comment by hlflanagan** [comment]
Discussed on [4 May 2026 meeting](https://github.com/w3c-fedid/meetings/blob/main/2026/2026-05-04-DCAPI-A-notes.md#aob). Should we just use (or point to) one of the diagrams at digitalcredentials.dev?

**Comment by marcoscaceres** [comment]
I’d like for us to own our own diagrams so we can keep them in sync if/when we update the spec. 

Let me get into this now…

**Comment by marcoscaceres** [comment]
Put up a PR for adding Mermaid support to ReSpec 
https://github.com/speced/respec/pull/5269

Playing around with various options. Not loving any of them 🫠 

This is not high priority, but will drop in a few suggestions for suggestions

**Comment by marcoscaceres** [comment]
Closing as I concluded we don't need this right now.
